modules/core/src/main/scala/ch/epfl/bluebrain/nexus/admin/core/resources/Resources.scala

```diff
@@ -120,31 +122,26 @@ class Resources[F[_], A: IdResolvable: TypeFilterExpr](agg: Agg[F], sparqlClient
     *
     * @param id     the identifier of the resource
     * @param rev    the last known revision of the resource instance
-    * @param tags   the tags added to the consequent [[ResourceEvent]] which might be created as a result of this operation
-    * @param persId the unique persistence ID of the resource
     * @return a [[RefVersioned]] instance wrapped in the abstract ''F[_]'' type
     *         if successful, or a [[ch.epfl.bluebrain.nexus.admin.core.Fault]] wrapped within ''F[_]'' otherwise
     */
-  def deprecate(id: A, rev: Long)(tags: Set[String], persId: String)(
-      implicit ctx: CallerCtx,
-      @silent perms: HasWriteProjects): F[RefVersioned[A]] =
-    evaluate(DeprecateResource(id, rev, ctx.meta, tags + persId), persId, s"Deprecate res '$id'")
+  def deprecate(id: A, rev: Long)(implicit ctx: CallerCtx): F[RefVersioned[A]] =
+    evaluate(DeprecateResource(id, rev, ctx.meta, tags + id.persistenceId), id.persistenceId, s"Deprecate res '$id'")
       .map(r => RefVersioned(id, r.rev))
 
   /**
     * Queries the system for the latest revision of the resource identified by the argument ''persId''.
     * The (in)existence of the resource is represented by the [[scala.Option]] type wrapped within the ''F[_]'' context.
     *
     * @param id     the identifier of the resource
-    * @param persId the unique persistence ID of the resource
     * @return an optional [[Resource]] instance wrapped in the
     *         abstract ''F[_]'' type if successful, or a [[ch.epfl.bluebrain.nexus.admin.core.Fault]] wrapped within
     *         ''F[_]'' otherwise
     */
-  def fetch(id: A)(persId: String)(implicit @silent perms: HasReadProjects): F[Option[Resource[A]]] =
-    agg.currentState(persId).map {
+  def fetch(id: A): F[Option[Resource[A]]] =
```

Comment:
you need an evidence that you can call that method, otherwise anybody can.
`(implicit @silent perms: HasReadProjects)`